### PR TITLE
feat: add `math/base/special/acotdf`

### PR DESCRIPTION
**Pull request by gunjjoshi**
## Description

> What is the purpose of this pull request?

This pull request:

-   adds `math/base/special/acotdf`, which would be the single-precision variant for [`math/base/special/acotd`](https://github.com/stdlib-js/stdlib/tree/develop/lib/node_modules/%40stdlib/math/base/special/acotd).

```c
float stdlib_base_acotdf( const float x );
```

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   resolves a part of #649.

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

### CODE REVIEW

**Review by Planeshifter — APPROVED** [review]
LGTM!

File: lib/node_modules/@stdlib/math/base/special/acotdf/benchmark/benchmark.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+// MAIN //
+
+bench( pkg, function benchmark( b ) {
+	var x;
+	var y;
+	var i;
+
+	b.tic();
+	for ( i = 0; i < b.iterations; i++ ) {
+		x = ( randu() * 2.0 ) - 1.0;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/acotdf/benchmark/benchmark.js]: @gunjjoshi Would you mind updating this and the other benchmark file?

File: lib/node_modules/@stdlib/math/base/special/acotdf/benchmark/c/native/benchmark.c [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+static double benchmark( void ) {
+	double elapsed;
+	double t;
+	float x;
+	float y;
+	int i;
+
+	t = tic();
+	for ( i = 0; i < ITERATIONS; i++ ) {
+		x = ( 2.0f * rand_float() ) - 1.0f;
```
Comment (kgryte) [inline lib/node_modules/@stdlib/math/base/special/acotdf/benchmark/c/native/benchmark.c]: Same comment.

**Review by kgryte — CHANGES_REQUESTED** [review]
Looks good. Once the benchmarks are updated, I'll approve and merge.

File: lib/node_modules/@stdlib/math/base/special/acotdf/benchmark/benchmark.js (hunk 1 above)
Comment (gunjjoshi) [reply lib/node_modules/@stdlib/math/base/special/acotdf/benchmark/benchmark.js]: Done!

**Review by kgryte — APPROVED** [review]
LGTM.

**Pull request merged**